service/history/shard/context.go

```diff
@@ -633,9 +638,6 @@ func (s *contextImpl) CreateWorkflowExecution(
 		return nil, err
 	}
 
-	if s.isClosed() {
```

Comment:
shard can be closed by another thread before this thread reaching line 620, so it's still reachable

Reply:
That's true! The description I made is wrong. Do you think we should not delete it? I only deleted it because of the new line check — it would be quite hard to write a test hitting this. That's of cause a terrible reason to make the code worse, just to be clear. 

I suppose it is better to have both? What do you think?